### PR TITLE
STRAT-64 & STRAT-25 Implement Variable Step and Stop/Turn Handling

### PR DESCRIPTION
**Pull request by wirooo**

### CODE REVIEW

File: optimization/optimizer.py [hunk 1]
```diff
@@ -3,14 +3,15 @@
...
+
 global car
 car = Car()
 
 parser = argparse.ArgumentParser(description="Add parameters to model")
-parser.add_argument("-d", "--distance", type=int, required=True, help="Distance to travel (m)")
-parser.add_argument("-t", "--time", type=int, required=True, help="Maximum allowable time (s)")
-parser.add_argument("-v", "--min_velocity", type=int, required=True, help="Minimum allowable velocity (m/s)")
-parser.add_argument("-s", "--step", type=int, required=True,help="Distance between elevation profile measurements")
+parser.add_argument("-d", "--distance", type=int, default=5490, help="Distance to travel (m)")
```
Comment (clarkevandenhoven) [inline optimization/optimizer.py]: Why did you remove the required?

File: optimization/optimizer.py [hunk 2]
```diff
@@ -36,48 +37,60 @@ def load_course_map(course_name="COTA"):
                 clean_data.append(row)
         # Create the elevation profile
         elev_profile = []
+        stops = []
         for i in range(len(clean_data) - 1):
-            pitch = numpy.arctan(float(clean_data[i][1]) / 10)
-            elev_profile.append(pitch)
+            pitch = numpy.arctan(float(clean_data[i][1]) / 10)  # is 10 = 30/3? i.e. rise over run?
```
Comment (clarkevandenhoven) [inline optimization/optimizer.py]: Yep. Should clean that up

File: optimization/optimizer.py [hunk 3]
```diff
@@ -36,48 +37,60 @@ def load_course_map(course_name="COTA"):
...
     """
     :param time: Maximum allowable time to cover a distance in seconds
     :param distance: Distance to be covered in meters
     :param e_profile: List of pitches the car must travel
+    :param min_velocity: Minimum allowable velocity
+    :param stop_profile: List of indices where car must stop
     """
-    avg_velocity = distance / time
-    dist_step = distance / len(e_profile)
+    avg_velocity = distance / time #should we have distance param? shouldnt we calculate from e/vprofile
```
Comment (clarkevandenhoven) [inline optimization/optimizer.py]: Will you should ask these questions in your own review instead of in comments

File: optimization/optimizer.py (hunk 1 above)
Comment (wirooo) [reply optimization/optimizer.py]: Required wouldn't allow the use of default values. Now when no parameters are entered it just uses the default instead.

File: optimization/optimizer.py [hunk 4]
```diff
@@ -3,14 +3,16 @@
...
 parser = argparse.ArgumentParser(description="Add parameters to model")
-parser.add_argument("-d", "--distance", type=int, required=True, help="Distance to travel (m)")
-parser.add_argument("-t", "--time", type=int, required=True, help="Maximum allowable time (s)")
-parser.add_argument("-v", "--min_velocity", type=int, required=True, help="Minimum allowable velocity (m/s)")
-parser.add_argument("-s", "--step", type=int, required=True,help="Distance between elevation profile measurements")
+parser.add_argument("-d", "--distance", type=int, default=5490, help="Distance to travel (m)")
+parser.add_argument("-t", "--time", type=int, default=450, help="Maximum allowable time (s)")
+parser.add_argument("-v", "--min_velocity", type=int, default=5, help="Minimum allowable velocity (m/s)")
+parser.add_argument("-s", "--step", type=int, default=30, help="Distance between elevation profile measurements")
+parser.add_argument("-st", "--stop_velocity", type=int, default=7, help="Maximum velocity for stops/turns")
```
Comment (wirooo) [inline optimization/optimizer.py]: Added user input for maximum stop velocity

File: optimization/optimizer.py [hunk 5]
```diff
@@ -36,48 +39,62 @@ def load_course_map(course_name="COTA"):
                 clean_data.append(row)
         # Create the elevation profile
         elev_profile = []
+        stops = []
+        total_dist = 0
         for i in range(len(clean_data) - 1):
-            pitch = numpy.arctan(float(clean_data[i][1]) / 10)
-            elev_profile.append(pitch)
+            pitch = numpy.arctan(float(clean_data[i][1]) / (float(clean_data[i][2])/ydir))
```
Comment (wirooo) [inline optimization/optimizer.py]: Cleaned up pitch calculation to adapt to variability

File: optimization/optimizer.py [hunk 6]
```diff
@@ -19,13 +21,14 @@ def load_course_map(course_name="COTA"):
...
     if course_name == "COTA":
-        with open('COTAelevation.txt', 'r') as file:
+        with open('COTAelevation_var.txt', 'r') as file:
             line = file.readline()
             count = 1
             data = []
             while line and count < 3:
                 count += 1
                 line = file.readline()
+            ydir = float(line.split(" ")[1][0])
```
Comment (wirooo) [inline optimization/optimizer.py]: In this implementation ydir does not change for the entire profile. Should it be different with each step, or is it supposed to be the same always.

File: optimization/optimizer.py (hunk 2 above)
Comment (wirooo) [reply optimization/optimizer.py]: Cleaned up in latest commit by adding ydir variability
